### DISCUSSION TITLE
建议补全 Props 相关属性，以便提供更好地自动补全

### DISCUSSION THREAD

**Discussion by tolking** [element-plus/element-plus#6658] [Ideas]
## 起因

我注意到如：

- InputProps 的类型描述中缺失 `maxlength` `minlength` `name` 等属性，虽然使用毫无问题。但类型不全 Volar 无法补全这些属性。
- AutocompleteProps 中部分属性与 InputProps 通用，但没有定义。而是通过 attrs 向 `el-input` 组件传值。
- InputProps 类型的可选值不全，如：`autocomplete` `type`

## 建议

- 或许可以补全所有属性和可选值
- 在定义 Props 时，或许可以使用 `pick` `omit` (lodash) 复用重复部分
- 在 setup 函数中可以通过 `reactivePick` `reactiveOmit` (vueuse) 拆分 props 替换 attrs

**Comment by sxzz** [comment]
对于问题一，`maxlength`, `minlength`, `name` 等这些属性是 attr，input 的原生属性。input 等元素的原生属性太多，不可能全部加上。目前只能看看 Volar 或社区有什么更好的方案。
